homeassistant/components/matter/lock.py

```diff
@@ -203,39 +259,27 @@ def _update_from_device(self) -> None:
             clusters.DoorLock.Attributes.LockState
         )
 
-        # always reset the optimisically (un)locking state on state update
-        self._reset_optimistic_state(write_state=False)
-
         LOGGER.debug("Lock state: %s for %s", lock_state, self.entity_id)
 
-        if lock_state == clusters.DoorLock.Enums.DlLockState.kUnlatched:
-            self._attr_is_locked = False
-            self._attr_is_open = True
-        elif lock_state == clusters.DoorLock.Enums.DlLockState.kLocked:
+        # If LockState reports a value of 0 - NotFullyLocked - ignore it as it is not a reliable indicator.
```

Comment:
Will I be able to detect NotFullyLocked from Home Assistant? This is important since my lock (Level Lock) does not expose alarms and I need to be able to determine this state of the lock.

Note that the Home Assistant definition of "jammed" is "incomplete locking", which maps almost exactly to NotFullyLocked:

https://github.com/home-assistant/core/blob/0cda0c449f7e20a5c9dbe29a97b19d7450d9650f/homeassistant/components/lock/__init__.py#L202-L204

If not, then I will still try to get my PR merged (#153610) after this one lands.

Reply:
> Will I be able to detect NotFullyLocked from Home Assistant? This is important since my lock (Level Lock) does not expose alarms and I need to be able to determine this state of the lock.

Odd that the lock doesn't expose alarms - are you sure? The DoorLockAlarm event is mandatory as is the specific LockJammed alarm code.